src/olympia/api/throttling.py

```diff
@@ -11,7 +13,23 @@
 
 
 class GranularUserRateThrottle(UserRateThrottle):
+    """
+    Throttling class that works like DRF's UserRateThrottle but supports
+    granular rates like 1/5second, and can be deactivated through a
+    API_THROTTLING django setting.
+
+    Its scope defaults to `user` but in most cases we'll want the child class
+    to override that and add a custom rate.
+    """
+
     RATE_REGEX = r'(?P<num>\d+)\/(?P<period_num>\d{0,2})(?P<period>\w)'
+    timer = time.time
+
+    def __init__(self):
+        super().__init__()
+        # Re-initialize timer at __init__() to allow freeze_gun.freeze_time()
+        # to work properly in tests.
+        self.timer = time.time
```

Comment:
But then you don't really need the class-property `timer` as subclasses will have to overwrite `__init__` anyway.

Reply:
I don't understand your point. Subclasses don't overwrite `__init__` - without that patch, they only have the class property, with it they have an instance one. That makes all the difference with `freeze_gun`.